controllers/activemqartemis_reconciler.go

```diff
@@ -1510,6 +1510,10 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) NewPodTemplateSpecForCR(customR
 		podSpec = &corev1.PodSpec{}
 	}
 
+	// Configure ImagePullSecrets if set on the DeploymentPlan
+	imagePullSecretRefs := pods.MakeImagePullSecretRefs(customResource.Spec.DeploymentPlan.ImagePullSecrets)
+	podSpec.ImagePullSecrets = imagePullSecretRefs
```

Comment:
If the array of string is replaced with an array of [LocalObjectReference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#localobjectreference-v1-core) it could just be assign to `podSpec.ImagePullSecrets`, i.e.
```
podSpec.ImagePullSecrets = customResource.Spec.DeploymentPlan.ImagePullSecrets
```